lib/ansible/executor/module_common.py

```diff
@@ -1331,14 +1335,36 @@ def get_action_args_with_defaults(action, args, defaults, templar):
     if module_defaults:
         module_defaults = templar.template(module_defaults)
 
-        # deal with configured group defaults first
-        if action in C.config.module_defaults_groups:
-            for group in C.config.module_defaults_groups.get(action, []):
-                tmp_args.update((module_defaults.get('group/{0}'.format(group)) or {}).copy())
+        if AnsibleCollectionRef.is_valid_fqcr(resolved_name):
+
+            collection_ref = AnsibleCollectionRef.from_fqcr(resolved_name, 'modules')
+
+            collection_name = collection_ref.collection
+            resource = collection_ref.resource
+
+            collection_routing = _get_collection_metadata(collection_name)
+            collection_groups = collection_routing.get('action_groups', {})
+            redirect_action_groups = collection_routing.get('action_groups_redirection', {})
+
+            action_groups = collection_groups.get(resolved_name, collection_groups.get(action, collection_groups.get(resource, [])))
+            for group in action_groups:
+                if AnsibleCollectionRef.is_valid_fqcr(group):
+                    fqcr = group
+                else:
+                    fqcr = '%s.%s' % (collection_name, group)
+
+                tmp_args.update((module_defaults.get('group/{0}'.format(fqcr)) or {}).copy())
+
+                redirected_groups = redirect_action_groups.get(fqcr, redirect_action_groups.get(group, []))
+
+                for redirected_group in redirected_groups:
+                    tmp_args.update((module_defaults.get('group/{0}'.format(redirected_group)) or {}).copy())
```

Comment:
When I look at your above example:
```.yaml
   action_groups_redirection:
     another_group:
       redirect: exact_match
```
then `redirected_groups` would be a `dict`, and `redirected_group` will be `redirect`. If I use
```.yaml
   action_groups_redirection:
     another_group:
       - exact_match
```
instead, it works.

Reply:
Fixed